java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -41,16 +41,16 @@ static synchronized void init() throws IOException {
   }
 
   /**
-   * Finds a named library in classpath resources, copies to temporary file on the filesystem to be
-   * loaded using {@link System#load}.
+   * Finds a named library in lib path. Falls back to classpath resources. In that case, it copies resource
+   * to temporary file on the filesystem to be loaded using {@link System#load}.
    *
    * @param library The bare name of the library.
    * @throws IOException If the file failed to read or write.
    */
   private static void load(String library) throws IOException {
```

Comment:
Is the argument a full path or relative path?

Reply:
The argument is the library name. System.loadLibrary looks at the configured lib path and it takes the library name. If that fails, then the mappedLibraryName is used to find a file in the java resources.

Reply:
Might be worth noting that this method calls `System.loadLibrary` in the first case, as you explicitly mention `System.load` for the latter.

Reply:
Just one question, what if there are multiple "onnxruntime.dll" in the the configured lib path? For example, if someone put a "onnxruntime.dll" in the "C:\Windows\System32" folder, what's the behavior?

Reply:
It's implementation dependent (i.e. JVM implementation dependent). I think OpenJDK will iterate the `java.library.path` and load the first one it finds, but I'm not sure.

Reply:
It's important to us to make it clear.  For example, what's the default value of "java.library.path" and what's the default behavioral if the user already have an "onnxruntime.dll" in "C:\Windows\System32", is it possible for him to use the file in the jar instead?

Reply:
I was about to update the docs to make it clearer, but I realized that I think I got the load order backwards. It should load from the classpath using the temp file and System.load() and _then_ fall back to the java.library.path and System.loadLibrary(). This ordering is so the specifically included one in the JAR on the classpath is loaded first before looking at the system. So it is more specific to more general. I could imagine somebody putting a version of onnxruntime on the lib path for one use case and then later using the Java API (choosing to use the library included in the JAR) and in that case it should look in the more specific place (the JAR) prior to looking in the more general place (system lib path). Thoughts? It is a simple change that seems to make logical sense. And of course I will update the javadoc regardless of what we decide.

Reply:
Loading from the classpath first doesn't allow me to use an immutable container unless you're very careful about catching the exceptions as the classpath load will fail with some kind of IOException. The old code did these things, if you supplied a runtime argument to the JVM then it loaded from java.library.path, otherwise by default it loaded from the classpath.

Reply:
I removed the fallback mechanism. In my experimental branch, the logic is now if-else: https://github.com/yuzawa-san/onnxruntime/blob/java-gradle-loader/java/src/main/java/ai/onnxruntime/OnnxRuntime.java#L56
What do y'all think of this? If you wanted to load from lib path, then you would omit the jars with the shared libraries. And there is no fallback. It is disjoint based on the presence of the resources. To use the lib path, you would not add the jar with the libraries to the class path.

Reply:
I'm a little wary of removing all control as users don't tend to be as in control of their classpath as we might like to think they are. If the onnx runtime is 3 nodes deep in their dependency graph (as it is in some projects I have), then the user relies upon two other library developers making the same decisions about including the native libraries jar or not, and they have no way of controlling that (especially if the jars are shaded).

However, I think that if you do it this way then you should catch the `UnsatisfiedLinkException` that comes out of this method and rethrow it with a nicer message that tells the user what needs to happen (i.e. they need to add the libraries jar to their classpath, or they need to add the libraries to `java.library.path`).

Reply:
I made the loader much more flexible by looking at properties. I'm hoping this would solve any issues that users may have. One property allows for the user to load their own libraries. Another property allows the user to specify an explicit filesystem path to the shared. library file. In the absence of the flags, then the logic checks the resources. If the resource is present then it is used otherwise the System.loadLibrary is used.

Reply:
I'd make the library names public static final constants on this class, as they are now part of the API. Also I'm unclear why this class became `public` instead of package private as it was before. If it's to make the Javadoc more visible then I recommend putting that javadoc in the `package-info.java` for `ai.onnxruntime`.

Reply:
done. I originally did make it public for the javadoc only.